### DISCUSSION TITLE
V4: Pre-select filters from URL (aka filters deep links in V4)

### DISCUSSION THREAD

**Discussion by nikosmsk** [microsoft-search/pnp-modern-search#1367] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently in V4 it's not possible to pre-select filters from URL parameters. This feature would be highly desirable.
Users should access the search page from external applications such as Dynamics 365 Sales and Dynamics 365 F&O.  A link must contain additional parameters such as CustomerID or QuotationID , to be able to show documents relevant to a given customer or quotation.  CustomerID and QuotationID are examples of refiners.   

**Describe the solution you'd like**
Implement filters deep links for V4: https://microsoft-search.github.io/pnp-modern-search/v3/search-parts/search-results/#filters-deep-links

**Describe alternatives you've considered**
Just use a url param q=mySearchTerm  and deal with users asking reasonable questions 'why this filter is not selected by default'.

**Additional context**
There were a couple of similar requests  (#860 #836) but all closed now.

**Comment by FranckyC** [comment]
Hi @nikosmsk, this is on the backlog. However I can't tell when it will be done. Since the filters schema has changed between v3 and v4 this is not a straightforward process unfortunately. For now, you can still use query parameters and the search box Web Part to configure an initial query.

**Comment by raymond-tishenko** [comment]
Hi @FranckyC  I'd like to add my vote to this item, it'd be a handy addition to v4, we're using it in v3 at the moment.

**Comment by mjarmstrong** [comment]
Hi @FranckyC, it looks like this was removed from the backlog / issues list. Was this intentional? This piece of functionality is preventing us from moving completely over to v4 (from v3). Thank you.

**Reply by wobba** [reply]
It's here in discussions as an enhancement idea. So not removed. Awaiting someone to have time and need to get it implemented. The feature is also related to #301.

**Comment by FranckyC** [comment]
Hi all, look at the PR https://github.com/microsoft-search/pnp-modern-search/pull/1432 adding this feature and let us know your feedbacks.
